dbt/adapters/sqlserver/connections.py

```diff
@@ -271,6 +272,40 @@ def get_pyodbc_attrs_before(credentials: SQLServerCredentials) -> Dict:
     return attrs_before
 
 
+def handle_datetimeoffset(dto_value) -> datetime:
+    """
+    Provides an pyodbc handler for DATETIMEOFFSET as this is not handled natively
+
+    Parameters
+    ----------
+    dto_value : Buffer
+        A binary DATETIMEOFFSET value
+
+    Returns
+    -------
+    out : datetime
+
+    Source
+    ------
+    pyodbc wiki:
+        https://github.com/mkleehammer/pyodbc/wiki/Using-an-Output-Converter-function
+
+    @stevenwinfield's decoding suggestion here:
+        https://github.com/mkleehammer/pyodbc/issues/134#issuecomment-281739794
+    """
+    tup = struct.unpack("<6hI2h", dto_value)  # e.g., (2017, 3, 16, 10, 35, 18, 500000000, -6, 0)
+    return datetime(
```

Comment:
Named arguments would make this code a little bit clearer

Reply:
Done